app/src/main/java/com/haigcramlin/cars/MainActivity.kt

```diff
@@ -1,31 +1,144 @@
+        var adapter =
+            ArrayAdapter<String>(this, R.layout.support_simple_spinner_dropdown_item, customList)
+        var adapter2 =
+            ArrayAdapter<String>(this, R.layout.support_simple_spinner_dropdown_item, customList)
+        choose_car.adapter = adapter
+        choose_car2.adapter = adapter2
+        choose_car.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onNothingSelected(parent: AdapterView<*>?) {
+            }
+
+            override fun onItemSelected(adapterView: AdapterView<*>?, view: View?, position: Int, id: Long) {
+                 car1 = adapterView?.getItemAtPosition(position).toString()
+                Toast.makeText(
+                    this@MainActivity,
+                    "Racer 1 selects: ${adapterView?.getItemAtPosition(position).toString()}",
+                    Toast.LENGTH_LONG
+                ).show()
+            }
+        }
+
+        choose_car2.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onNothingSelected(parent: AdapterView<*>?) {}
+            override fun onItemSelected(adapterView: AdapterView<*>?, view: View?, position: Int, id: Long) {
+                car2 = adapterView?.getItemAtPosition(position).toString()
+                Toast.makeText(
+                    this@MainActivity,
+                    "Racer 2 selects: ${adapterView?.getItemAtPosition(position).toString()}",
+                    Toast.LENGTH_LONG
+                ).show()
+            }
+        }
+    }
+
+    fun createCarsAndTracks() {
         val ferrari = Car("ferrari", "testarossa")
         val honda = Car("honda", "speed wagon")
         val honda2 = Car("honda", "civic")
+        val ford = Car("ford", "fiesta")
+        val chevy = Car("corvette", "stingray")
+
         val speed500 = Racetrack("SPEED500")
         val radiatorSprings = Racetrack("Radiator Springs")
         val leader = LeaderBoard()
-        val ford = Car("ford", "fiesta")
-        val chevy = Car("corvette", "stingray")
+    }
+
+    fun setUpButton() {
+        button.setOnClickListener {
+            raceSelectedCars()
+        }
+    }
+
+    fun addSpinnerListener(spinner:Spinner, isCar1: Boolean) {
+        spinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onNothingSelected(parent: AdapterView<*>?) {}
+
+            override fun onItemSelected(adapterView: AdapterView<*>?, view: View?, position: Int, id: Long)  {
+                if(isCar1)
+                    car1 = adapterView?.getItemAtPosition(position).toString()
+                else
+                    car2 = adapterView?.getItemAtPosition(position).toString()
+                var homey: List<String> = arrayListOf(car1.toString(), car2.toString())
+
+            }
+        }
+    }
+
+    fun createCars() {
+
+    }
+
+    fun getCarlist() {
+
+    }
+
+    fun drawableResource(winner: String?) {
```

Comment:
Might want to rename this fun to be something more descriptive. Drawable resource is an android term and this could be confused with other actions/properties. I would suggest something like setWinnerDrawable() or updateDrawableForWinner() or something.

Reply:
kk, I've changed it to updateDrawableForWinner()